dask/layers.py

```diff
@@ -409,11 +388,18 @@ def __init__(
         # much earlier.
         #
         # See https://github.com/dask/dask/pull/6051 for a detailed discussion.
-        self.annotations = self.annotations or {}
-        if "priority" not in self.annotations:
-            self.annotations["priority"] = {}
-        self.annotations["priority"]["__expanded_annotations__"] = None
-        self.annotations["priority"].update({_key: 1 for _key in self.get_split_keys()})
+        annotations = annotations or {}
+
+        if "priority" not in annotations:
+            _splits = self.get_split_keys()
+
+            def _set_prio(key):
+                if key in _splits:
+                    return 1
+                return 0
+
+            annotations["priority"] = _set_prio
+        super().__init__(annotations=annotations)
```

Comment:
I find the annotations API a bit unintuitive. I would've expected a mapping like

`{"key": {"priority": 42}}`

but instead it is a 

`{"priority": {"key": 42}}`

More or less it's the same thing but I find myself more naturally iterating over keys than over annotations